components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/component/ClientManager.java

```diff
@@ -0,0 +1,229 @@
+     * @param payload The payload to include in the request body.
+     * @return A configured HttpPost instance.
+     * @throws HTTPAdapterException If an error occurs while creating the request.
+     */
+    public HttpPost createHttpPost(String url, Object payload, String secret) throws HTTPAdapterException {
+
+        HttpPost request = new HttpPost(url);
+        request.setHeader(ACCEPT, ContentType.APPLICATION_JSON.getMimeType());
+        request.setHeader(CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
+
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+
+        String jsonString;
+        try {
+            jsonString = mapper.writeValueAsString(payload);
+            request.setEntity(new StringEntity(jsonString));
+        } catch (IOException e) {
+            throw HTTPAdapterUtil.handleClientException(ERROR_PUBLISHING_EVENT_INVALID_PAYLOAD);
+        }
+
+        // Add HMAC signature header if secret is provided
+        if (secret != null && !secret.isEmpty()) {
+            try {
+                String signature = "sha256=" + hmacSha256Hex(secret, jsonString);
+                request.setHeader(X_HUB_SIGNATURE, signature);
+            } catch (Exception e) {
+                throw HTTPAdapterUtil.handleClientException(ERROR_CREATING_HMAC_SIGNATURE);
+            }
+        }
+
+        return request;
+    }
+
+    // Utility method for HMAC SHA-256 hex encoding
+    private static String hmacSha256Hex(String secretKey, String data) throws NoSuchAlgorithmException,
+            InvalidKeyException {
+
+        Mac mac = Mac.getInstance(HMAC_SHA256_ALGORITHM);
+        SecretKeySpec key =
+                new SecretKeySpec(secretKey.getBytes(StandardCharsets.UTF_8),
+                        HMAC_SHA256_ALGORITHM);
+        mac.init(key);
+        byte[] rawHmac = mac.doFinal(data.getBytes(StandardCharsets.UTF_8));
+        StringBuilder sb = new StringBuilder();
+        for (byte b : rawHmac) {
+            sb.append(String.format("%02x", b));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Execute an HTTP POST request asynchronously.
+     *
+     * @param httpPost The HTTP POST request to execute.
+     * @return A CompletableFuture containing the HTTP response.
+     */
+
+    public CompletableFuture<HttpResponse> executeAsync(HttpPost httpPost) {
+
+        CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+
+        getHttpAsyncClient().execute(httpPost, new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(HttpResponse result) {
+
+                future.complete(result);
+            }
+
+            @Override
+            public void failed(Exception ex) {
+
+                future.completeExceptionally(new IdentityRuntimeException("Execution exception", ex));
```

Comment:
Throw a meaningful error and ideally we should handle connection and read timeouts and implement retry mechanism

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/37/commits/15538dc2b9ce588c47e8db9c5db39156de457b3c